### PR TITLE
Truco/gas/misc1

### PR DESCRIPTION
**Pull request by 0xtruco**
### Purpose
Change many misc issues relating to gas opt

code-423n4/2021-12-yetifinance-findings#79
code-423n4/2021-12-yetifinance-findings#85
code-423n4/2021-12-yetifinance-findings#89
code-423n4/2021-12-yetifinance-findings#102
code-423n4/2021-12-yetifinance-findings#103
code-423n4/2021-12-yetifinance-findings#125
code-423n4/2021-12-yetifinance-findings#132
code-423n4/2021-12-yetifinance-findings#133
code-423n4/2021-12-yetifinance-findings#139
code-423n4/2021-12-yetifinance-findings#164
code-423n4/2021-12-yetifinance-findings#175
code-423n4/2021-12-yetifinance-findings#175
code-423n4/2021-12-yetifinance-findings#261
code-423n4/2021-12-yetifinance-findings#279
code-423n4/2021-12-yetifinance-findings#281
code-423n4/2021-12-yetifinance-findings#261
code-423n4/2021-12-yetifinance-findings#299
code-423n4/2021-12-yetifinance-findings#294

### Open Questions and Pre-Merge TODOs

- [ ]

### Testing Methodology
`npx hardhat compile`

### Learning

### CODE REVIEW

File: packages/contracts/contracts/AssetWrappers/WJLP.sol [hunk 1]
```diff
@@ -88,7 +88,7 @@ contract WJLP is ERC20_8, IWAsset {
         _symbol = ERC20_symbol;
         _name = ERC20_name;
         _decimals = ERC20_decimals;
-        _totalSupply = 0;
+        // _totalSupply = 0; default value
```
Comment (manuelreydias) [inline packages/contracts/contracts/AssetWrappers/WJLP.sol:91]: Remove this line

File: packages/contracts/contracts/Dependencies/LiquityMath.sol [hunk 2]
```diff
@@ -27,7 +27,7 @@ library LiquityMath {
     function decMul(uint x, uint y) internal pure returns (uint decProd) {
         uint prod_xy = x.mul(y);
 
-        decProd = prod_xy.add(DECIMAL_PRECISION / 2).div(DECIMAL_PRECISION);
+        decProd = prod_xy.add(5e17).div(DECIMAL_PRECISION);
```
Comment (manuelreydias) [inline packages/contracts/contracts/Dependencies/LiquityMath.sol:30]: Why is a non-named constant used? If this is correct, let's name `5e17` as a uint constant

File: packages/contracts/contracts/Dependencies/YetiCustomBase.sol [hunk 3]
```diff
@@ -160,17 +160,20 @@ contract YetiCustomBase is BaseMath {
...
+                if (coll3.amounts[i] != 0) {
+                    diffTokens[j] = coll3.tokens[i];
+                    diffAmounts[j] = coll3.amounts[i];
+                    j++;
+                }
             }
         }
         finalColls.tokens = diffTokens;
         finalColls.amounts = diffAmounts;
+        // returns finalColls;
```
Comment (manuelreydias) [inline packages/contracts/contracts/Dependencies/YetiCustomBase.sol:176]: Remove comment

File: packages/contracts/contracts/TroveManagerRedemptions.sol [hunk 4]
```diff
@@ -688,7 +688,7 @@ contract TroveManagerRedemptions is TroveManagerBase, ITroveManagerRedemptions {
     function _requireAfterBootstrapPeriod() internal view {
         uint256 systemDeploymentTime = yetiTokenContract.getDeploymentStartTime();
         require(
-            block.timestamp >= systemDeploymentTime.add(BOOTSTRAP_PERIOD),
+            block.timestamp >= systemDeploymentTime + BOOTSTRAP_PERIOD, // won't overflow
```
Comment (manuelreydias) [inline packages/contracts/contracts/TroveManagerRedemptions.sol:691]: Remove comment

File: packages/contracts/contracts/YUSDToken.sol [hunk 5]
```diff
@@ -238,15 +238,15 @@ contract YUSDToken is CheckContract, IYUSDToken {
         require(account != address(0), "_mint: account is address(0)");
 
         _totalSupply = _totalSupply.add(amount);
-        _balances[account] = _balances[account].add(amount);
+        _balances[account] = _balances[account] + amount; // can't overflow since total supply didn't
```
Comment (manuelreydias) [inline packages/contracts/contracts/YUSDToken.sol:241]: Remove comment

**Review by manuelreydias — APPROVED** [review]
Why are all the > changed to !=

**Comment by 0xtruco** [comment]
Apparently != is more gas efficient than >

**Pull request merged**